http-server-netty/src/main/java/io/micronaut/http/server/netty/types/files/NettySystemFileCustomizableResponseType.java

```diff
@@ -108,7 +114,33 @@ public ChannelFuture write(HttpRequest<?> request, MutableHttpResponse<?> respon
 
         if (response instanceof NettyMutableHttpResponse) {
 
-            NettyMutableHttpResponse nettyResponse = ((NettyMutableHttpResponse) response);
+            NettyMutableHttpResponse<?> nettyResponse = ((NettyMutableHttpResponse<?>) response);
+
+            // Parse the range headers (if any), and determine the position and content length
+            // Only `bytes` ranges are supported. Only single ranges are supported. Invalid ranges fall back to returning the full response.
+            // See https://httpwg.org/specs/rfc9110.html#field.range
+            long fileLength = getLength();
+            String rangeHeader = request.getHeaders().get(HttpHeaders.RANGE);
+            long position = 0;
+            long contentLength = fileLength;
+            if (rangeHeader != null
+                && request.getMethod() == HttpMethod.GET // A server MUST ignore a Range header field received with a request method that is unrecognized or for which range handling is not defined.
+                && rangeHeader.startsWith(UNIT_BYTES) // An origin server MUST ignore a Range header field that contains a range unit it does not understand.
+                && response.status() == HttpStatus.OK // The Range header field is evaluated after evaluating the precondition header fields defined in Section 13.1, and only if the result in absence of the Range header field would be a 200 (OK) response.
+            ) {
+                IntRange range = parseRangeHeader(rangeHeader, fileLength);
+                if (range != null // A server that supports range requests MAY ignore or reject a Range header field that contains an invalid ranges-specifier (Section 14.1.1)
+                    && range.firstPos < range.lastPos // A server that supports range requests MAY ignore a Range header field when the selected representation has no content (i.e., the selected representation's data is of zero length).
+                ) {
+                    position = range.firstPos;
+                    contentLength = range.lastPos + 1 - range.firstPos;
+                    response.status(HttpStatus.PARTIAL_CONTENT);
+                    response.header(CONTENT_RANGE, String.format("%s %d-%d/%d", UNIT_BYTES, range.firstPos, range.lastPos, fileLength));
```

Comment:
only respond with user-controlled data if file range is valid (see above)

Reply:
Good point, thanks! Added test cases for outside-of-file range requests, and also a couple for non-numeric range requests, and fixed the code.